### PR TITLE
Add prefetch partitions functional test

### PR DESCRIPTION
**Pull request by LiubovDidkivska**
Functional tests to check api compatibility.
Check full stack from network layer to user api.

Relates-To: OLPEDGE-2343

Signed-off-by: Liubov Didkivska <ext-liubov.didkivska@here.com>

### CODE REVIEW

File: tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientPrefetchPartitionsTest.cpp [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    EXPECT_TRUE(mock_server_client_->Verify());
+  }
+}
+
+TEST_F(VersionedLayerClientPrefetchPartitionsTest, PrefetchPartitionFail) {
+  olp::client::HRN hrn(kTestHrn);
+
+  auto client = dr::VersionedLayerClient(hrn, kLayer, boost::none, *settings_);
+  {
+    SCOPED_TRACE("Empty request");
```
Comment (andescu) [inline tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientPrefetchPartitionsTest.cpp:181]: This is not necessary as it is the same test already done in integration test. Functional tests is the "e2e" test that covers the entire pipeline from user API to network layer. So test only those that reach the network and require an answer from the server.

E.g. 200 partitions, 1 query fails the other succeeds.

File: tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientPrefetchTest.cpp [hunk 2]
```diff
@@ -18,23 +18,27 @@
  */
 
 #include <gtest/gtest.h>
```
Comment (andescu) [inline tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientPrefetchTest.cpp:20]: gtest.h is not a system include it is a project include.

**Review by andescu — COMMENTED** [review]
Where is the rest of the tests?

**Comment by LiubovDidkivska** [comment]
> Where is the rest of the tests?

updated. More tests were added.

**Review by andescu — APPROVED** [review]

**Review by andescu — APPROVED** [review]

File: tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientPrefetchTest.cpp [hunk 3]
```diff
@@ -51,28 +55,39 @@ class VersionedLayerClientPrefetchTest : public ::testing::Test {
...
     mock_server_client_.reset();
   }
+  std::string GetPartitions() {
+    return "/query/v1/catalogs/" + kTestHrn + "/layers/" + kLayer +
+           "/partitions";
+  }
+  std::string GenerateGetDataPath(const std::string& data_handle) {
+    return "/blob/v1/catalogs/" + kTestHrn + "/layers/" + kLayer + "/data/" +
+           data_handle;
+  }
```
[the remark this answers is not in the document]
Comment (LiubovDidkivska) [reply tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientPrefetchTest.cpp]: changed to use PlatformUrlsGenerator

File: tests/common/ApiDefaultResponses.h [hunk 4]
```diff
@@ -29,17 +29,16 @@ namespace mockserver {
 
 class ApiDefaultResponses {
  public:
-  static olp::client::Apis GenerateResourceApisResponse(
-      std::string catalog) {
+  static olp::client::Apis GenerateResourceApisResponse(std::string catalog) {
```
Comment (andescu) [inline tests/common/ApiDefaultResponses.h:32]: Why did you change the versions? `notification`, `publish` and `stream` are v2, see https://developer.here.com/documentation/data-api/api-reference.html. You need to provide exactly the same numbers we expect in the APIs, else our requests will not work.

**Review by andescu — DISMISSED** [review]

**Review by andescu — APPROVED** [review]

**Review by kzvieriev — APPROVED** [review]
